### PR TITLE
Get rid of Modular in interface names

### PR DESCRIPTION
**Pull request by lisitsyn**
It must be failing somewhere but lets see.

### CODE REVIEW

**Comment by lisitsyn** [comment]
@karlnapf checks are green but codecov, take a look at the patch please.

File: .travis.yml [hunk 1]
```diff
@@ -25,48 +25,48 @@ matrix:
...
       env:
-        - CMAKE_OPTIONS="-DOctaveModular=ON -DTRAVIS_DISABLE_UNIT_TESTS=ON -DTRAVIS_DISABLE_LIBSHOGUN_TESTS=ON"
-        - OCTAVE_MODULAR=true
+        - CMAKE_OPTIONS="-DOctave=ON -DTRAVIS_DISABLE_UNIT_TESTS=ON -DTRAVIS_DISABLE_LIBSHOGUN_TESTS=ON"
+        - OCTAVE=true
     - compiler: clang
       services: docker
       env:
-        - CMAKE_OPTIONS="-DRModular=ON -DTRAVIS_DISABLE_UNIT_TESTS=ON -DTRAVIS_DISABLE_LIBSHOGUN_TESTS=ON"
+        - CMAKE_OPTIONS="-DR=ON -DTRAVIS_DISABLE_UNIT_TESTS=ON -DTRAVIS_DISABLE_LIBSHOGUN_TESTS=ON"
```
Comment (vigsterkr) [inline .travis.yml]: `DR` is a bit.... mmmm but yeah why not.

**Comment by vigsterkr** [comment]
btw once i merge this, it'll break the whole interface/notebook generation on buildbot. @lisitsyn do you wanna fix the infra scripts for the buildbot as well? :)

**Comment by lisitsyn** [comment]
yes as well as debian

File: .travis.yml (hunk 1 above)
Comment (lisitsyn) [inline .travis.yml]: I thought of RLang or #rstats (the twitter hashtag for R). Is it better?

Comment (vigsterkr) [inline .travis.yml]: RLang is +1

File: .travis.yml [hunk 2]
```diff
@@ -25,48 +25,48 @@ matrix:
     - compiler: clang
       services: docker
       env:
-        - CMAKE_OPTIONS="-DPythonModular=ON -DTRAVIS_DISABLE_UNIT_TESTS=ON -DTRAVIS_DISABLE_LIBSHOGUN_TESTS=ON"
+        - CMAKE_OPTIONS="-DPython=ON -DTRAVIS_DISABLE_UNIT_TESTS=ON -DTRAVIS_DISABLE_LIBSHOGUN_TESTS=ON"
```
Comment (karlnapf) [inline .travis.yml]: This will make buildbot fail. 
Need to update the infra repo as well

File: .travis.yml (hunk 1 above)
Comment (karlnapf) [inline .travis.yml]: RLang

File: CMakeLists.txt [hunk 3]
```diff
@@ -87,24 +87,24 @@ SET(THIRD_PARTY_DIR ${CMAKE_SOURCE_DIR}/third_party)
...
-SET(OctaveModularDescription "Octave")
-SET(JavaModularDescription "Java")
-SET(PerlModularDescription "Perl")
-SET(RubyModularDescription "Ruby")
-SET(CSharpModularDescription "C#")
-SET(RModularDescription "R")
-SET(LuaModularDescription "Lua")
-SET(ScalaModularDescription "Scala")
+SET(AVAILABLE_INTERFACES Python;Octave;Java;Perl;Ruby;CSharp;R;Lua;Scala)
+SET(PythonDescription "Python")
```
Comment (karlnapf) [inline CMakeLists.txt]: Can we please name those "INTERFACE_PYTHON" etc, since we are touching them already.
There is an issue for that somewhere

Comment (karlnapf) [inline CMakeLists.txt]: and that is more inline with the other switches, they are also grouped when ordered alphabetically and autocomplete is easier

File: .travis.yml (hunk 2 above)
Comment (lisitsyn) [inline .travis.yml]: I know

**Pull request merged**